tests/subsys/pm/power_domain/src/main.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <pm/device.h>
+#include <pm/device_runtime.h>
+
+static int testing_domain_on_notitication;
+static int testing_domain_off_notitication;
+
+#define TEST_DOMAIN DT_NODELABEL(test_domain)
+#define TEST_DEVA DT_NODELABEL(test_dev_a)
+#define TEST_DEVB DT_NODELABEL(test_dev_b)
+
+static const struct device *domain, *deva, *devb;
+
+static int dev_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 0;
+}
+
+static int pm_action(const struct device *dev,
+		     enum pm_device_action pm_action)
+{
+	ARG_UNUSED(dev);
+
+	if (testing_domain_on_notitication > 0) {
+		if (pm_action == PM_DEVICE_ACTION_HANDLE_PD_RESUME) {
+			testing_domain_on_notitication--;
+		}
+	} else if (testing_domain_off_notitication > 0) {
+		if (pm_action == PM_DEVICE_ACTION_HANDLE_PD_SUSPEND) {
+			testing_domain_off_notitication--;
+		}
+	}
+
+	return 0;
+}
+
+PM_DEVICE_DT_DEFINE(TEST_DOMAIN, pm_action);
+DEVICE_DT_DEFINE(TEST_DOMAIN, dev_init, PM_DEVICE_DT_REF(TEST_DOMAIN),
+		 NULL, NULL, POST_KERNEL, 10, NULL);
+
+PM_DEVICE_DT_DEFINE(TEST_DEVA, pm_action);
+DEVICE_DT_DEFINE(TEST_DEVA, dev_init, PM_DEVICE_DT_REF(TEST_DEVA),
+		 NULL, NULL, POST_KERNEL, 20, NULL);
+
+PM_DEVICE_DT_DEFINE(TEST_DEVB, pm_action);
+DEVICE_DT_DEFINE(TEST_DEVB, dev_init, PM_DEVICE_DT_REF(TEST_DEVB),
+		 NULL, NULL, POST_KERNEL, 30, NULL);
+
+/**
+ * @brief Test the power domain behavior
+ *
+ * Scenarios tested:
+ *
+ * - get + put multiple devices under a domain
+ * - notification when domain state changes
+ */
+static void test_power_domain_device_runtime(void)
+{
+	int ret;
+	enum pm_device_state state;
+
+	domain = DEVICE_DT_GET(TEST_DOMAIN);
+	zassert_not_null(domain, "Failed to get device");
+
+	deva = DEVICE_DT_GET(TEST_DEVA);
+	zassert_not_null(deva, "Failed to get device");
+
+	devb = DEVICE_DT_GET(TEST_DEVB);
+	zassert_not_null(devb, "Failed to get device");
+
+	pm_device_runtime_enable(domain);
+	pm_device_runtime_enable(deva);
+	pm_device_runtime_enable(devb);
```

Comment:
nit: returns error code, may be asserted (or ignored with `(void)` if failure is not possible)

Reply:
In this pr revision `pm_device_runtime_enable` is still a void function. Will change it when rebasing ..